vyper/parser/parser_utils.py

```diff
@@ -891,22 +891,23 @@ def annotate_ast(
     RewriteUnarySubVisitor().visit(parsed_ast)
 
 
-def zero_pad(bytez_placeholder, maxlen, context=None, zero_pad_i=None):
-    zero_padder = LLLnode.from_list(['pass'])
-    if maxlen > 0:
-        # Iterator used to zero pad memory.
-        if zero_pad_i is None:
-            zero_pad_i = context.new_placeholder(BaseType('uint256'))
-        zero_padder = LLLnode.from_list([
-            'repeat', zero_pad_i, ['mload', bytez_placeholder], ceil32(maxlen), [
-                'seq',
-                # stay within allocated bounds
-                ['if', ['gt', ['mload', zero_pad_i], ceil32(maxlen)], 'break'],
-                ['mstore8', ['add', ['add', 32, bytez_placeholder], ['mload', zero_pad_i]], 0]
-            ]],
+def zero_pad(bytez_placeholder):
+    # calldatacopy from past-the-end gives zero bytes.
```

Comment:
Can you add a link to the section of the Yellowpaper that states this?

Reply:
see https://github.com/ethereum/vyper/pull/1605/commits/9daf153dbab72ecf45272b49e9734c0a6bde6568